x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/fields/ExtractedFields.java

```diff
@@ -87,8 +87,14 @@ protected ExtractedField detect(String field) {
                             : ExtractedField.ExtractionMethod.SOURCE;
                 }
             }
-            if (isFieldOfType(field, "geo_point") || isFieldOfType(field, "geo_shape")) {
-                return ExtractedField.newGeoField(field, internalField, method);
+            if (isFieldOfType(field, "geo_point")) {
+                if (method != ExtractedField.ExtractionMethod.DOC_VALUE) {
```

Comment:
I know I suggested checking this here and I'm sorry. I think this belongs in `GeoPointField`'s extractor.